### PR TITLE
Allow .xml in map names

### PR DESCRIPTION
**Pull request by axc**
Resolves https://github.com/battlecode/battlecode-server/issues/235

### CODE REVIEW

File: src/main/battlecode/server/Server.java [hunk 1]
```diff
@@ -275,7 +275,10 @@ private Team runMatch(GameInfo currentGame,
                           long[][] teamMemory,
                           ProxyWriter proxyWriter) throws Exception {
 
-        final String mapName = currentGame.getMaps()[matchIndex];
+        String mapName = currentGame.getMaps()[matchIndex];
+        if (mapName.endsWith(".xml")) {
```
Comment (kazimuth) [inline src/main/battlecode/server/Server.java]: This logic should be in GameMapIO.

**Comment by axc** [comment]
Updated.

File: src/main/battlecode/world/GameMapIO.java [hunk 2]
```diff
@@ -43,9 +43,12 @@
      */
     public static GameMap loadMap(String mapName, File mapDir)
             throws IOException {
-
         final GameMap result;
 
+        if (mapName.endsWith(".xml")) {
+            mapName = mapName.substring(0, mapName.indexOf('.'));
```
Comment (trun) [inline src/main/battlecode/world/GameMapIO.java]: `mapName.lastIndexOf('.')`?

**Comment by axc** [comment]
Updated again according to comments, and added unit test

**Pull request merged**